core/tee/tee_svc.c

```diff
@@ -897,12 +897,15 @@ TEE_Result tee_svc_get_time(enum utee_time_category cat, TEE_Time *mytime)
 		break;
 	}
 
-	if (res == TEE_SUCCESS || res == TEE_ERROR_OVERFLOW) {
-		res2 = tee_svc_copy_to_user(s, mytime, &t, sizeof(t));
-		if (res2 != TEE_SUCCESS)
-			res = res2;
+	if (res != TEE_SUCCESS && res != TEE_ERROR_OVERFLOW) {
+		t.seconds = 0;
+		t.millis = 0;
 	}
 
+	res2 = tee_svc_copy_to_user(s, mytime, &t, sizeof(t));
```

Comment:
If an error other than TEE_ERROR_OVERFLOW is returned, this structure is filled with zeroes.